jerry-core/debugger/jerry-sha1.c

```diff
@@ -0,0 +1,369 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ *  FIPS-180-1 compliant SHA-1 implementation
+ *
+ *  Copyright (C) 2006-2015, ARM Limited, All Rights Reserved
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  This file is part of mbed TLS (https://tls.mbed.org)
+ */
+/*
+ *  The SHA-1 standard was published by NIST in 1993.
+ *
+ *  http://www.itl.nist.gov/fipspubs/fip180-1.htm
+ */
+
+#ifdef JERRY_DEBUGGER
+
+#include "jerry-debugger.h"
+
+/**
+ * SHA-1 context structure.
+ */
+typedef struct
+{
+  uint32_t total[2]; /**< number of bytes processed */
+  uint32_t state[5]; /**< intermediate digest state */
+  uint8_t buffer[64]; /**< data block being processed */
+} jerry_sha1_context;
+
+/* 32-bit integer manipulation macros (big endian). */
+
+#define GET_UINT32_BE(n, b, i) \
+{ \
+  (n) = (((uint32_t) (b)[(i) + 0]) << 24) \
+        | (((uint32_t) (b)[(i) + 1]) << 16) \
+        | (((uint32_t) (b)[(i) + 2]) << 8) \
+        | ((uint32_t) (b)[(i) + 3]); \
+}
+
+#define PUT_UINT32_BE(n, b, i) \
+{ \
+  (b)[(i) + 0] = (uint8_t) ((n) >> 24); \
+  (b)[(i) + 1] = (uint8_t) ((n) >> 16); \
+  (b)[(i) + 2] = (uint8_t) ((n) >> 8); \
+  (b)[(i) + 3] = (uint8_t) ((n)); \
+}
+
+/**
+ * Initialize SHA-1 context.
+ */
+static void
+jerry_sha1_init (jerry_sha1_context *sha1_context_p) /**< SHA-1 context */
+{
+  memset (sha1_context_p, 0, sizeof (jerry_sha1_context));
+
+  sha1_context_p->total[0] = 0;
+  sha1_context_p->total[1] = 0;
+
+  sha1_context_p->state[0] = 0x67452301;
```

Comment:
The SHA-1 standard is mentioned in the first non-license block comment of the source file. I don't see the point of repeating it here. We could repeat it in every line of the code, since we invented nothing new here. I think the purpose (setting the initial state) is quite obvious.